pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -479,15 +479,8 @@ func buildGatewayNetworkFiltersFromTCPRoutes(node *model.Proxy, env *model.Envir
 		for _, tcp := range vsvc.Tcp {
 			if l4MultiMatch(tcp.Match, server, gatewaysForWorkload) {
 				upstream = tcp.Route[0].Destination // We pick first destination because TCP has no weighted cluster
-				destSvc, present := push.ServiceByHostname[model.Hostname(upstream.Host)]
-				if !present {
-					log.Debugf("service %q does not exist in the registry", upstream.Host)
-					return nil
-				}
-
-				return buildOutboundNetworkFilters(env, node,
-					istio_route.GetDestinationCluster(upstream, destSvc, int(server.Port.Number)),
-					"", port)
+				clusterName := istio_route.GetDestinationCluster(upstream, push.ServiceByHostname[model.Hostname(upstream.Host)], int(server.Port.Number))
```

Comment:
what does this return if the service does not exist?
`push.ServiceByHostname[model.Hostname(upstream.Host)]`
and how does `istio_route.GetDestinationCluster(...)` handle that value?

Reply:
that function has checks for nil services..
https://github.com/istio/istio/pull/9092/files#diff-209da7e6575ffc42b68cd8da72331384R235

It returns `outbound|listenerPort|destination.Subset|destination.Host` always. the service is used only to get the default port if none is specified in the destination.